x-pack/plugins/files/public/components/image/image.tsx

```diff
@@ -28,22 +45,59 @@ export interface Props extends ImgHTMLAttributes<HTMLImageElement> {
  * ```
  */
 export const Image = React.forwardRef<HTMLImageElement, Props>(
-  ({ src, alt, onFirstVisible, ...rest }, ref) => {
+  (
+    { src, alt, onFirstVisible, onLoad, onError, meta, wrapperProps, size = 'original', ...rest },
+    ref
+  ) => {
+    const [isLoaded, setIsLoaded] = useState<boolean>(false);
+    const [blurDelayExpired, setBlurDelayExpired] = useState(false);
     const { isVisible, ref: observerRef } = useViewportObserver({ onFirstVisible });
+
+    useEffect(() => {
+      const id = window.setTimeout(() => setBlurDelayExpired(true), 200);
+      return () => window.clearTimeout(id);
+    }, []);
```

Comment:
Before executing React state change in async method, would be great to check if component is still mounted:

```suggestion
    useEffect(() => {
      let unmounted = false;
      const id = window.setTimeout(() => if(!unmounted) setBlurDelayExpired(true), 200);
      return () => {
        unmounted = true;
        window.clearTimeout(id);
      };
    }, []);
```